util/util_test.go

```diff
@@ -1,10 +1,217 @@
 package util
 
 import (
+	"fmt"
 	"testing"
 	"time"
 )
 
+func TestNewTimePeriod(t *testing.T) {
+	timezone, err := time.LoadLocation("Australia/Brisbane")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, tc := range []struct {
+		from     time.Time
+		to       time.Time
+		expected TimePeriod
+	}{
+		// when it's already normalized
+		{
+			time.Date(0, 0, 0, 15, 04, 05, 06, time.UTC),
+			time.Date(0, 0, 0, 16, 04, 05, 06, time.UTC),
+			TimePeriod{
+				From: time.Date(0, 0, 0, 15, 04, 05, 06, time.UTC),
+				To:   time.Date(0, 0, 0, 16, 04, 05, 06, time.UTC),
+			},
+		},
+		// it normalizes to very first day
+		{
+			time.Date(1869, 9, 24, 15, 04, 05, 06, time.UTC),
+			time.Date(1869, 9, 24, 16, 04, 05, 06, time.UTC),
+			TimePeriod{
+				From: time.Date(0, 0, 0, 15, 04, 05, 06, time.UTC),
+				To:   time.Date(0, 0, 0, 16, 04, 05, 06, time.UTC),
+			},
+		},
+		// it ignores the timezone
+		{
+			time.Date(1869, 9, 24, 15, 04, 05, 06, timezone),
+			time.Date(1869, 9, 24, 16, 04, 05, 06, timezone),
+			TimePeriod{
+				From: time.Date(0, 0, 0, 15, 04, 05, 06, time.UTC),
+				To:   time.Date(0, 0, 0, 16, 04, 05, 06, time.UTC),
+			},
+		},
+	} {
+		got := NewTimePeriod(tc.from, tc.to)
+		if tc.expected.From != got.From {
+			t.Fatalf("expected %v, got %v", tc.expected, got)
+		}
+		if tc.expected.To != got.To {
+			t.Fatalf("expected %v, got %v", tc.expected, got)
+		}
+	}
+}
+
+func TestTimePeriodIncludes(t *testing.T) {
+	atTheMoment := NewTimePeriod(time.Now().Add(-1*time.Minute), time.Now().Add(+1*time.Minute))
+
+	midnight := NewTimePeriod(
+		time.Date(1869, 9, 23, 23, 00, 00, 00, time.UTC),
+		time.Date(1869, 9, 24, 01, 00, 00, 00, time.UTC),
+	)
```

Comment:
Please correct me if I'm wrong, but I believe this is not a valid test because your actual code in `ParseTimePeriods` does not account for the beginning and ending times spanning midnight, so it would never actually produce the pair of `begin` and `end` times you are generating here. For instance, you have no test that correctly handles the `excluded-times-of-day` you give in the README: `22:00-08:00`, which spans midnight. If you want to support that you have to detect that the end time is "earlier" than the begin time and add 24 hours to it. But beware, this will break during Daylight Saving nights and it will be very difficult to do this correctly (besides defining what "correct" is in that case). This is why I avoided supporting suspension periods spanning midnight. :-)

Reply:
Right, I'll definitely double-check that but I think it worked (apart from the daylight savings issue).

The idea is to use `time.Parse` to convert `strings` to `time` in the same manner as go does; however we don't care about the actual day and strip it off later. I believe the test above works with any day other than the 24th as well.

So this will end up in a time range that looks like `start: 23, end: 1` (on day zero). `start` being greater than `end` idicates that the range goes across midnight which is triggers a different logic branch for whether a time is included or not in `Includes()`: https://github.com/linki/chaoskube/pull/62/files#diff-99f38c7c07bfcc03a528180716bafab5R37

Reply:
Regarding daylight savings: If `time.Parse` returns the `hours` that would be shown on the clock, e.g. on start it jumps from `01:59` to `03:00` and on end it jumps from `02:59` to `02:00`, then I think this should also work. Of course there will be one real hour more or less chaos time, but I think that's fine.

Reply:
Ah! Missed that bit of logic in `Includes()`. :+1:

Reply:
> Regarding daylight savings: ...

Makes sense. I just think it's important that you satisfy yourself that you have adequately addressed the potential issue. :+1: